torch_xla/csrc/tensor.cpp

```diff
@@ -619,16 +619,18 @@ void XLATensor::SetXlaData(xla::ComputationClient::DataPtr xla_data,
   }
 }
 
-void XLATensor::SetIrValue(ir::Value ir_value) {
+void XLATensor::SetIrValue(ir::Value ir_value, bool inplace) {
   data()->xla_data = nullptr;
   data()->tensor_data = c10::nullopt;
-  if (data()->view != nullptr) {
-    // If we have an active view, and a SetIrValue() happens, it means we are
-    // within an in-place execution context, and we need to update the view's
+  if (data()->view != nullptr && inplace) {
+    // If we have an active view, SetIrValue() happens, and we are
+    // within an in-place execution context, we need to update the view's
     // alias as well.
     data()->view = UpdateView(data()->view, std::move(ir_value));
     data()->generation += 1;
   } else {
+    // Reset the view if we are not within an in-place execution context
+    data()->view = nullptr;
```

Comment:
Technically do we need to reset `data()->generation` to zero too? Maybe that doesn't end up mattering in practice though.

Reply:
right, I will reset that as well